### ISSUE TITLE
Training setup + Validation results

### ISSUE THREAD

**Issue by SiMoM0** [huixiancheng/SENet#31]
Hi,
thanks for this great work.

I have a question about the training setup for SemanticKITTI to train from scratch the models with image resolutions $(64\times1024)$ and $(64\times2048)$. Is it the same of the model with images $(64\times512)$ ?

Moreover, could you please provide the results on the validation set of the models with higher image resolutions ?

Thanks

**Comment by huixiancheng** [comment]
Hi @SiMoM0 
For Q1:  Yes. You can keep the same lr setting. For me, the train setting just limited by GPU Cost. So train 30 epochs for 1024&2048. You can also adjust it as it is not optimized by searching.
For Q2：I‘m not sure your meaning. 1024&2048 model has been released. Maybe you can get better performance with higher resolution like Tornado-net and KPRNet.

**Comment by SiMoM0** [comment]
Ok, so the only way to train the models with images $(64\times1024)$ and $(64\times2048)$ is to use the pretrained model with resolution $(64\times512)$. Otherwise, I would not be able to replicate the results reported in the paper, correct?

Oh, thank you for clarifying that. I had assumed they were related to SalsaNext based on the name.

**Comment by nlgranger** [comment]
Same here, I was not able to reproduce the results in the paper with fine tuning.
At the end of the 64x512 cycle I get around 61mIoU on validation which is apparently normal. But then 63.3 with 1024x64 and 62.8 with 2048x64.

Strange thing is, when resuming the model is better initially (checkpoint trained on half resolution) than at the end of the first epoch. Is the lr_schedule still cosine with an initial ramp during fine-tuning ?

**Comment by IKUNLOVER-FAN** [comment]
> Same here, I was not able to reproduce the results in the paper with fine tuning. At the end of the 64x512 cycle I get around 61mIoU on validation which is apparently normal. But then 63.3 with 1024x64 and 62.8 with 2048x64.
> 
> Strange thing is, when resuming the model is better initially (checkpoint trained on half resolution) than at the end of the first epoch. Is the lr_schedule still cosine with an initial ramp during fine-tuning ?
Hi,
thanks for this great work.
1、May I ask how can I obtained the result of Article 64.7 MIOU? I followed the readme provided by the author, but no matter what, I couldn't achieve it, and the effect was only around 62. May I ask if you need any additional strategies and training techniques? Can you teach me?
2、If you use validation sets to optimize the model to improve accuracy, what are the specific optimization operations. Or other training techniques, could you please let me know?
We look forward to your reply
Best

**Comment by huixiancheng** [comment]
Hi~@IKUNLOVER-FAN  Thanks for your attention and report. Sry for long time passed. I forgot some details.
For finetune, I just merge train and val set here and use low lr (e.g. 1e-5? sry i forget it) train 3~5 epochs. 
https://github.com/huixiancheng/CENet/blob/9a84103d186a1f93637cae3d96426760deb04140/modules/trainer.py#L63-L65
However, this is an intuitive setup and trick. Unable to find optimal policy. Probably improve 1.5-2 miou on the test set.
Maybe you could find better policy in other newly SOTA papers.

**Comment by nlgranger** [comment]
Thank you very much for the follow up.
BTW your code is pleasure to work with when coming from the abomination that is mmdetection ;-) !

**Comment by MrTang886** [comment]
Hello, thank you for your good work. I have two questions for you:
1. May I ask whether lr is still a cosine in the process of fine tuning, or should we use the 1e-5 fixed learning rate you mentioned? I used the cosine learning rate in the original text to keep the same and trained 5 epochs, but the accuracy of the test set after online submission is three or four points lower than that before I did not fine tuning. If the fixed learning rate is used, How exactly does the code need to be adjusted?
2. The original article did not use test time enhancement. If I wanted to use test time enhancement, what code would I need to add where?
Look forward to your reply, thank you!

> Hi~@IKUNLOVER-FAN Thanks for your attention and report. Sry for long time passed. I forgot some details. For finetune, I just merge train and val set here and use low lr (e.g. 1e-5? sry i forget it) train 3~5 epochs.
> 
> https://github.com/huixiancheng/CENet/blob/9a84103d186a1f93637cae3d96426760deb04140/modules/trainer.py#L63-L65
> 
> 
> However, this is an intuitive setup and trick. Unable to find optimal policy. Probably improve 1.5-2 miou on the test set.
> Maybe you could find better policy in other newly SOTA papers.

**Comment by huixiancheng** [comment]
For Q1: maybe you could try comment code here https://github.com/huixiancheng/CENet/blob/9a84103d186a1f93637cae3d96426760deb04140/modules/trainer.py#L510 here.
For Q2: you could try follow 2dpass/frnet or other paper. vote for all tta infer results could get better metrics.

**Comment by MrTang886** [comment]
Ok, thank you for your reply!
